db/compaction/compaction_picker.cc

```diff
@@ -39,7 +39,14 @@ bool FindIntraL0Compaction(const std::vector<FileMetaData*>& level_files,
                            size_t min_files_to_compact,
                            uint64_t max_compact_bytes_per_del_file,
                            uint64_t max_compaction_bytes,
-                           CompactionInputFiles* comp_inputs) {
+                           CompactionInputFiles* comp_inputs,
+                           SequenceNumber oldest_mem_seqno) {
+
+  // Do not pick ingested file when there is at least one memtable not flushed which of seqno is overlap with the sst.
+  if (level_files[0]->fd.smallest_seqno == level_files[0]->fd.largest_seqno &&
```

Comment:
Would it be better to remove this conditional? Is there ever a case where we want to pick an sstable for intra-L0 compaction that has a largest seqno that is larger than the smallest unflushed seqno?

Reply:
Yes. I have explained it in my description.  If the sstable was produced by ingested, and its key was not overlap with memtable, it would use the global seqno, which is just in [smallest seqno, largest seqno] of memtable.